File: hnaj-be/app/Exceptions/VisitException.php
<?php

namespace App\Exceptions;

use App\Enums\VisitErrorCode;
use RuntimeException;

/**
 * Lỗi nghiệp vụ của domain visit.
 * Mang sẵn HTTP status và mã lỗi để handler dựng error envelope chung.
 */
class VisitException extends RuntimeException
{
    public function __construct(
        public readonly VisitErrorCode $errorCode,
        string $message,
        public readonly int $status = 422,
    ) {
        parent::__construct($message);
    }
}

File: hnaj-be-java/src/main/java/com/hnaj/api/exception/VisitException.java
package com.hnaj.api.exception;

public class VisitException extends RuntimeException {

    private final VisitErrorCode errorCode;
    private final int status;

    public VisitException(VisitErrorCode errorCode, String message, int status) {
        super(message);
        this.errorCode = errorCode;
        this.status = status;
    }

    public VisitErrorCode getErrorCode() {
        return errorCode;
    }

    public int getStatus() {
        return status;
    }
}
